### PR TITLE
Update configfiles.md

### PR DESCRIPTION
**Pull request by samsp-msft**
Goal: Make it easy for someone who is not that familiar with asp.net to understand the options available in config.

Q. (from Tratcher) How do we keep it up to date with code changes
A. The same as other documentation - by hand - unless we want to start adding something custom to /// comments, and the doc builder, but even then it would need manual work.

### CODE REVIEW

File: docs/docfx/articles/configfiles.md [hunk 1]
```diff
@@ -83,3 +83,123 @@ The proxy will apply the given matching criteria and policies, and then pass off
...
+## All config properties
+```
+{
+  // Base URLs the server listens on, must be configured independently of the routes below
+  "Urls": "http://localhost:6000;https://localhost:6001",
+  "Logging": {
+    "LogLevel": {
+      "Default": "Information",
+      // Uncomment to hide diagnostic messages from runtime and proxy
+      // "Microsoft": "Warning",
```
Comment (samsp-msft) [inline docs/docfx/articles/configfiles.md:96]: Will probably need to change based on namespace changes

File: docs/docfx/articles/configfiles.md [hunk 2]
```diff
@@ -83,3 +83,123 @@ The proxy will apply the given matching criteria and policies, and then pass off
...
+        }
+      },
+      "allclusterprops": {
+        "Destinations": {
+          "first_destination": {
+            "Address": "https://contoso.com"
+          },
+          "another_destination": {
+            "Address": "https://10.20.30.40",
+            "Health" : "https://10.20.30.40:12345/test" // override for active health checks
```
Comment (samsp-msft) [inline docs/docfx/articles/configfiles.md:158]: One thing that wasn't clear from health check docs, what is health endpoint supposed to expose?

File: docs/docfx/articles/configfiles.md [hunk 3]
```diff
@@ -83,3 +83,123 @@ The proxy will apply the given matching criteria and policies, and then pass off
...
+          }
+        },
+        "LoadBalancingPolicy" : "PowerOfTwoChoices", // Alternatively "First", "Random", "RoundRobin", "LeastRequests"
+        "SessionAffinity": {
+          "Enabled": "true", // defaults to 'false'
```
Comment (samsp-msft) [inline docs/docfx/articles/configfiles.md:163]: do we use quotes for true/false?

File: docs/docfx/articles/configfiles.md [hunk 4]
```diff
@@ -83,3 +83,123 @@ The proxy will apply the given matching criteria and policies, and then pass off
...
+          "Mode": "Cookie", // default, alternatively "CustomHeader"
+          "FailurePolicy": "(Redistribute", // default, Alternatively "Return503"
```
Comment (samsp-msft) [inline docs/docfx/articles/configfiles.md:165]: Remove (

File: docs/docfx/articles/configfiles.md [hunk 5]
```diff
@@ -83,3 +83,123 @@ The proxy will apply the given matching criteria and policies, and then pass off
...
+          "Passive": { // Disables destinations based on HTTP response codes
+            "Enabled": true, //Defaults to false
+            "Policy" : "TransportFailureRateHealthPolicy", // Required
+            "ReactivationPeriod" : "00:00:10" // 10s
+          }
+        },
+        "HttpClient" : { // Configuration of HttpClient instance used to contact destinations
+          "SSLProtocols" : "Tls13",
+          "DangerousAcceptAnyServerCertificate" : false,
+          "ClientCertificate" : "?? How does this look in config??",
```
Comment (samsp-msft) [inline docs/docfx/articles/configfiles.md:187]: @wfurt what is the syntax for this?

File: docs/docfx/articles/configfiles.md [hunk 6]
```diff
@@ -83,3 +83,123 @@ The proxy will apply the given matching criteria and policies, and then pass off
...
+          "MaxConnectionsPerServer" : 1024,
+          "ActivityContextHeaders" : "???",
```
Comment (samsp-msft) [inline docs/docfx/articles/configfiles.md:189]: @karelz, @scalablecory  Syntax?

File: docs/docfx/articles/configfiles.md (hunk 2 above)
Comment (Tratcher) [reply docs/docfx/articles/configfiles.md:158]: The health endpoint should return a 2xx response. That's all we check for at the moment.
https://github.com/microsoft/reverse-proxy/blob/7d26dffc565ccaca7c44a46002e819df1a8b18b2/src/ReverseProxy/Service/HealthChecks/ConsecutiveFailuresHealthPolicy.cs#L62

File: docs/docfx/articles/configfiles.md (hunk 5 above)
Comment (Tratcher) [reply docs/docfx/articles/configfiles.md:187]: This test is a good reference for all of the properties:
https://github.com/microsoft/reverse-proxy/blob/7d26dffc565ccaca7c44a46002e819df1a8b18b2/test/ReverseProxy.Tests/Configuration/ConfigurationConfigProviderTests.cs#L209-L216
Note for this property there are sets of options that go together.
For a file you do:
```
                    ""Path"": ""mycert.pfx"",
                    ""KeyPath"": null,
                    ""Password"": ""myPassword1234"",
```
For loading from the cert store you do:
```
                    ""Subject"": null,
                    ""Store"": null,
                    ""Location"": null,
                    ""AllowInvalid"": null
```

File: docs/docfx/articles/configfiles.md (hunk 6 above)
Comment (Tratcher) [reply docs/docfx/articles/configfiles.md:189]: Here's another good reference:
https://github.com/microsoft/reverse-proxy/blob/7d26dffc565ccaca7c44a46002e819df1a8b18b2/src/ReverseProxy/Configuration/ConfigurationConfigProvider.cs#L333
This one's an enum.

File: docs/docfx/articles/configfiles.md (hunk 2 above)
Comment (samsp-msft) [reply docs/docfx/articles/configfiles.md:158]: Added that to the health article

File: docs/docfx/articles/configfiles.md (hunk 3 above)
Comment (samsp-msft) [reply docs/docfx/articles/configfiles.md]: parsing was happy without quotes, so going with that.

**Comment by samsp-msft** [comment]
@Tratcher - ready for submission

File: docs/docfx/articles/configfiles.md [hunk 7]
```diff
@@ -83,3 +83,135 @@ The proxy will apply the given matching criteria and policies, and then pass off
...
+            "Health" : "https://10.20.30.40:12345/test" // override for active health checks
+          }
+        },
+        "LoadBalancingPolicy" : "PowerOfTwoChoices", // Alternatively "First", "Random", "RoundRobin", "LeastRequests"
+        "SessionAffinity": {
+          "Enabled": true, // Defaults to 'false'
+          "Mode": "Cookie", // Default, alternatively "CustomHeader"
+          "FailurePolicy": "Redistribute", // default, Alternatively "Return503"
+          "Settings" : {
+              "CustomHeaderName": "MySessionHeaderName" // Defaults to 'X-Microsoft-Proxy-Affinity`
```
Comment (Tratcher) [inline docs/docfx/articles/configfiles.md:168]: I've just updated this in my other PR.
```suggestion
              "CustomHeaderName": "MySessionHeaderName" // Defaults to 'X-Yarp-Proxy-Affinity`
```

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Looks ok, but you need to resolve the merge conflict.

**Comment by samsp-msft** [comment]
Should fix #751.

**Review by Tratcher — APPROVED** [review]

**Pull request merged**